plugins/ssrmatching/ssrmatching.ml

```diff
@@ -773,10 +772,10 @@ let rec uniquize = function
   if max_occ <= !nocc then p', u.up_dir, (sigma, uc, u.up_t)
   else errorstrm (str"Only " ++ int !nocc ++ str" < " ++ int max_occ ++
         str(String.plural !nocc " occurrence") ++ match upats_origin with
-        | None -> str" of" ++ spc() ++ pr_constr_pat p'
+        | None -> str" of" ++ spc() ++ pr_constr_pat (Global.env () (* XXX *)) sigma p'
```

Comment:
@gares which `env` should I use here and below?

Reply:
uff, you don't have it at hand. It is the env the companion closure gets in input. You could make this function take an env rather than a unit, but the the user would need to pass the same env to both (that is the proof env).

I'm tempted to say that you could change the find_P type, make a pair adding an env, return it in the first closure, and bind it in the second one.

The old code was using the global one, that seems a but to me.

Reply:
> The old code was using the global one, that seems a but to me.

@gares Does it mean my patch preserves the former semantics here? That's the main goal I'm trying to achieve.